`Lift Text` instance is now available in Data.Text causing compilation failure

mustache fails to compile against [Data.Text 1.2.4.0](http://hackage.haskell.org/package/text-1.2.4.0/changelog) because this latest release (uploaded August 10, 2019) includes its own `Lift Text` instance.

The compilation error is: 

```
[3 of 8] Compiling Text.Mustache.Internal.Types ( src/Text/Mustache/Internal/Types.hs, dist/build/Text/Mustache/Internal/Types.o )

src/Text/Mustache/Internal/Types.hs:350:10: error:
    Duplicate instance declarations:
      instance Lift Text
        -- Defined at src/Text/Mustache/Internal/Types.hs:350:10
      instance Lift Text -- Defined in ‘Data.Text’
    |
350 | instance Lift Text where
    |          ^^^^^^^^^
```

mustache should probably remove this instance and add a version dependency on this specific version of `Data.Text`.

--- a/src/Text/Mustache/Internal/Types.hs
+++ b/src/Text/Mustache/Internal/Types.hs
@@ -1,5 +1,6 @@
 {-# OPTIONS_GHC -fno-warn-orphans  #-}
 {-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE CPP                        #-}
 {-# LANGUAGE FlexibleInstances          #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE MultiParamTypeClasses      #-}
@@ -347,8 +348,11 @@ data Template = Template
 instance Lift TemplateCache where
   lift m = [| HM.fromList $(lift $ HM.toList m) |]
 
+--Data.Text 1.2.4.0 introduces its own Lift Text instance
+#if !MIN_VERSION_text(1,2,4)
 instance Lift Text where
   lift = lift . unpack
+#endif
 
 deriveLift ''DataIdentifier
 deriveLift ''Node

